README.md

```diff
@@ -160,21 +143,104 @@ try {
 }
 ```
 
+Some kinds of dependency fallbacks may be handled by [import maps](https://github.com/wicg/import-maps), but even with the support of import maps, [some scenarios](https://github.com/WICG/import-maps/blob/master/README.md#alternating-logic-based-on-the-presence-of-a-built-in-module) which are outside of the declaratively handled set would still benefit from top-level `await`.
+
+### WebAssembly Modules
+
+WebAssembly Modules are "compiled" and "instantiated" in a logically asynchronous way, based on their imports: Some WebAssembly implementations do nontrivial work at either phase, which is important to be able to shunt off into another thread. To integrate with the JavaScript module system, they will need to do the equivalent of a top-level await. See the [WebAssembly ESM integration proposal](https://github.com/webassembly/esm-integration) for more details.
+
+## Semantics as desugaring
+
+Top-level `await` makes importing a module automatically block on any top-level `await`s. You can think of it like, each module exports a Promise, and after all the `import` statements, but before the rest of the module, the Promises are all `await`ed:
```

Comment:
I might approach this from a different direction (or, start out with a different explanation). Something like:

Currently, a module waits for all of its dependencies to execute all of their statements before the import is considered finished, and the module's code can run. While introducing `await`, we want to keep this property: dependencies should still execute through to the end, even if you need to wait for that execution to finish asynchronously. The natural way to think of this is as if each module exported a promise...